### PR TITLE
benchmark: ensure all watcher receivers to finish

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5099.

### CODE REVIEW

**Comment by gyuho** [comment]
Previous code has blocking events sent to `results` channel, and causing panic after we do `close(results)`. Fix by:

1. give more buffer to `results` channel
2. wait until all watch receivers to finish (previous code only waits for one of watch receiver to finish, I think)


I got it to reproduce that issue, but don't see the `send to closed channel` panic anymore.

/cc @heyitsanthony @xiang90

File: tools/benchmark/cmd/watch.go [hunk 1]
```diff
@@ -142,15 +139,14 @@ func watchFunc(cmd *cobra.Command, args []string) {
...
-	results = make(chan result)
+	results = make(chan result, 1000)
 	bar = pb.New(eventsTotal)
 
 	bar.Format("Bom !")
 	bar.Start()
 
 	atomic.StoreInt32(&nrRecvCompleted, 0)
 	recvCompletedNotifier = make(chan struct{})
-	close(putStartNotifier)
```
Comment (gyuho) [inline tools/benchmark/cmd/watch.go]: I delete this because we already have `watchCompletedNotifier`?

File: tools/benchmark/cmd/watch.go [hunk 2]
```diff
@@ -142,15 +139,14 @@ func watchFunc(cmd *cobra.Command, args []string) {
 	// put phase
 	// total number of puts * number of watchers on each key
 	eventsTotal = watchPutTotal * (watchTotal / watchedKeyTotal)
-	results = make(chan result)
+	results = make(chan result, 1000)
```
Comment (heyitsanthony) [inline tools/benchmark/cmd/watch.go]: this seems more like a hack than a genuine fix...

Comment (gyuho) [inline tools/benchmark/cmd/watch.go]: right. let me delete this part.

File: tools/benchmark/cmd/watch.go [hunk 3]
```diff
@@ -169,7 +165,9 @@ func watchFunc(cmd *cobra.Command, args []string) {
 		close(putreqc)
 	}()
 
-	<-recvCompletedNotifier
+	for range streams {
+		<-recvCompletedNotifier
```
Comment (heyitsanthony) [inline tools/benchmark/cmd/watch.go]: Also change `recvWatchChan` so `recvCompletedNotifier <- struct{}{}` happens outside the `range wch` loop?

Comment (gyuho) [inline tools/benchmark/cmd/watch.go]: `atomic.AddInt32(&nrRecvCompleted, 1)` happens inside `for` loop, so I think it should be inside `for` loop?

**Comment by gyuho** [comment]
Just removed `buffer`. PTAL. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
thanks.

**Pull request merged**